### ISSUE TITLE
Does the description of the "heif_init" functions is correct? [QUESTION]

### ISSUE THREAD

**Issue by bigcat88** [strukturag/libheif#909]
```
// For backwards compatibility, it is not really necessary to call heif_init(), but if you don't, the plugins
// registered by default may not be freed correctly.
```

With latest versions from PPA without calling `heif_init` I do not see any loaded plugins.
When call it, all works as expected.
Should the description be corrected?

**Comment by farindk** [comment]
Yes, the text is easy to misunderstand. Without `heif_init`, no external plugins are loaded. But there may also be plugins that are directly compiled into the library. These are never freed. There are also other things that are not freed without `heif_deinit`.
I'll change the text.

**Comment by bigcat88** [comment]
Thanks for quick reply, feel free to close issue as resolved :)

**Comment by farindk** [comment]
Thanks for the report.

**Comment by fancycode** [comment]
> Yes, the text is easy to misunderstand. Without `heif_init`, no external plugins are loaded. But there may also be plugins that are directly compiled into the library. These are never freed. There are also other things that are not freed without `heif_deinit`. I'll change the text.

Maybe it would be good to always load external plugins on first use, even if `heif_init` was not called. For most applications it doesn't really matter that some memory will not be explicitly released on shutdown. Especially with the packaging now building all codecs as plugins, simply upgrading the packages might break existing applications as codecs suddently will not be available any more (see #912).

**Comment by farindk** [comment]
Yes, that should be changed in the next release.
